lib/debug/thread_client.rb

```diff
@@ -367,7 +367,11 @@ def frame_eval src, re_raise: false
 
       result = if b
```

Comment:
I think `TracePoint.allow_reentry` should wrap the entire evaluation part. That is:

```rb
def eval_with_binding(b, src)
  if b
    f, _l = b.source_location
    b.eval(src, "(rdbg)/#{f}")
  else
    frame_self = current_frame.self
    instance_eval_for_cmethod(frame_self, src)
  end
end

# in frame_eval
result = if TracePoint.respond_to?(:allowallow_reentry)
           TracePoint.allowallow_reentry do
             eval_with_binding(b, src)
           end
         else
           eval_with_binding(b, src)
         end
```

Reply:
Done.